src/filter.ts

```diff
@@ -156,5 +144,5 @@ class CompositeFilter extends EntityFilter {
 }
 
 export function isFilter(filter: any): filter is EntityFilter {
-  return (filter as EntityFilter).toProto !== undefined;
+  return filter instanceof EntityFilter;
```

Comment:
why does this have to change?

Reply:
I suggested the fix so I'll answer.

`filter.toProto !== undefined` only checks if the object has a `toProto` properties (not set to `undefined`)
It is the case that `EntityFilter`s have a `toProto` methods but it is not necessarily the case that an object with a `toProto` property is an `EntityFilter`.

`filter instanceof EntityFilter` is the correct check for `filter is EntityFilter`

I hope that makes sense.